pkg/download/classic/download_test.go

```diff
@@ -364,3 +377,74 @@ func TestDownload_MalformedResponseFromAnAPI(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Len(t, configurations, 1)
 }
+
+func TestDownload_SkippedParentsSkipChildren(t *testing.T) {
+	c := dtclient.NewMockClient(gomock.NewController(t))
+	c.EXPECT().ListConfigs(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, a api.API) ([]dtclient.Value, error) {
+		if a.ID == "PARENT_API_ID" {
+			return []dtclient.Value{{Id: "PARENT_ID_1", Name: "PARENT_NAME_1"}}, nil
+		}
+		require.FailNow(t, "Unexpected API ID")
+		return nil, nil
+	}).Times(2)
+
+	parentAPI := api.API{
+		ID:            "PARENT_API_ID",
+		URLPath:       "PARENT_API_PATH",
+		NonUniqueName: true}
+
+	apiMap := api.APIs{
+		"PARENT_API_ID": parentAPI,
+		"CHILD_API_ID": api.API{ID: "CHILD_API_ID",
+			URLPath:       "CHILD_API_PATH",
+			NonUniqueName: false,
+			Parent:        &parentAPI}}
+
+	contentFilters := map[string]ContentFilter{
+		"PARENT_API_ID": {
+			ShouldBeSkippedPreDownload: func(value dtclient.Value) bool { return true },
+		},
+	}
+	configurations, err := Download(c, "project", apiMap, contentFilters)
+	require.NoError(t, err)
+	assert.Len(t, configurations, 0, "Expected no configurations as everything is skipped")
+}
+
+func TestDownload_SingleConfigurationChild(t *testing.T) {
+	c := dtclient.NewMockClient(gomock.NewController(t))
+	c.EXPECT().ListConfigs(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, a api.API) ([]dtclient.Value, error) {
+		if a.ID == "PARENT_API_ID" {
+			return []dtclient.Value{{Id: "PARENT_ID_1", Name: "PARENT_NAME_1"}}, nil
+		}
+		require.FailNow(t, "Unexpected API ID")
+		return nil, nil
+	}).Times(2)
+
+	c.EXPECT().ReadConfigById(gomock.Any(), gomock.Any()).Return([]byte("{}"), nil).AnyTimes()
+
+	parentAPI := api.API{
+		ID:            "PARENT_API_ID",
+		URLPath:       "PARENT_API_PATH",
+		NonUniqueName: true}
+
+	apiMap := api.APIs{
+		"PARENT_API_ID": parentAPI,
+		"CHILD_API_ID": api.API{ID: "CHILD_API_ID",
+			URLPath:             "CHILD_API_PATH",
+			NonUniqueName:       false,
+			Parent:              &parentAPI,
+			SingleConfiguration: true}}
+
+	contentFilters := map[string]ContentFilter{
+		"PARENT_API_ID": {
+			ShouldBeSkippedPreDownload: func(value dtclient.Value) bool { return false },
+		},
+	}
+
+	configurations, err := Download(c, "project", apiMap, contentFilters)
+	require.NoError(t, err)
+	require.Len(t, configurations, 2, "Expected two configurations")
+	require.Len(t, configurations["PARENT_API_ID"], 1)
+	require.Len(t, configurations["CHILD_API_ID"], 1)
+	assert.Equal(t, configurations["PARENT_API_ID"][0].Coordinate.ConfigId+configurations["PARENT_API_ID"][0].Coordinate.ConfigId, configurations["CHILD_API_ID"][0].Coordinate.ConfigId, "Single child config should have the same config ID as parent")
```

Comment:
This is the one thing I'm not really happy with: Currently the child config ID is the parent ID concatenated to itself, see https://github.com/Dynatrace/dynatrace-configuration-as-code/blob/52f30ebbb945362f92ee89ecb9a7308678bc6291/pkg/download/classic/download.go#L312. Ideally the `value` would know it originated from a single configuration here.

Reply:
Yes, that works for me 😄 . There may also be follow up chances to use `ID()` - particularly in logs - e.g. other places in `pkg/download/classic/download.go`